src/app/modules/time-clock/store/entry.effects.ts

```diff
@@ -21,7 +21,11 @@ export class EntryEffects {
         map((response) => {
           const stopDateForEntry = new Date(response.end_date);
           stopDateForEntry.setSeconds(stopDateForEntry.getSeconds() + 1 );
-          return new actions.CreateEntry({ project_id: action.idProjectSwitching, start_date: stopDateForEntry.toISOString() });
+          return new actions.CreateEntry({
+            project_id: action.idProjectSwitching,
+            start_date: stopDateForEntry.toISOString(),
```

Comment:
I know this is not your change, but it seems something is wrong here:
**start_date**: **stopDateForEntry**.toISOString(),

Reply:
This has to do with the logic of clocking out, when switching a project. The stopDate of the entry being clocked out is the startDate of the new time-entry.

Reply:
Yep, I get what you mean. In that case I would change line 23 to this:
`const newTimeEntryStartDate = stopDateForEntry.setSeconds(stopDateForEntry.getSeconds() + 1 );`
and line  26 to:
`start_date: newTimeEntryStartDate.toISOString(),`

In that way, it is easier to read.